ludwig/hyperopt/execution.py

```diff
@@ -585,6 +588,103 @@ def execute(
         return hyperopt_results
 
 
+class RayTuneExecutor(HyperoptExecutor):
+    def __init__(self, hyperopt_sampler, output_feature, metric, split, goal, **kwargs):
+        ray.init(ignore_reinit_error=True)
+        HyperoptExecutor.__init__(self, hyperopt_sampler, output_feature,
+                                  metric, split)
+        self.search_space = hyperopt_sampler
+        self.output_feature = output_feature
+        self.metric = metric
+        self.split = split
+        self.goal = goal
+        self.trial_id = 0
+        self.experiment_kwargs = {}
+
+    def _run_experiment(self, config):
+
+        self.trial_id += 1
+        hyperopt_dict = copy.deepcopy(self.experiment_kwargs)
+        modified_config = substitute_parameters(
+            copy.deepcopy(hyperopt_dict["config"]), config)
+        hyperopt_dict["config"] = modified_config
+        hyperopt_dict["experiment_name"] = f'{hyperopt_dict["experiment_name"]}_{self.trial_id}'
+
+        train_stats, eval_stats = run_experiment(**hyperopt_dict)
+        metric_score = self.get_metric_score(eval_stats)
+
+        tune.report(parameters=str(config), metric_score=metric_score,
+                    training_stats=str(train_stats), eval_stats=str(eval_stats))
+
+    def execute(self,
+                config,
+                dataset=None,
+                training_set=None,
+                validation_set=None,
+                test_set=None,
+                training_set_metadata=None,
+                data_format=None,
+                experiment_name="hyperopt",
+                model_name="run",
+                # model_load_path=None,
+                # model_resume_path=None,
+                skip_save_training_description=False,
+                skip_save_training_statistics=False,
+                skip_save_model=False,
+                skip_save_progress=False,
+                skip_save_log=False,
+                skip_save_processed_input=False,
+                skip_save_unprocessed_output=False,
+                skip_save_predictions=False,
+                skip_save_eval_stats=False,
+                output_directory="results",
+                gpus=None,
+                gpu_memory_limit=None,
+                allow_parallel_threads=True,
+                use_horovod=None,
+                random_seed=default_random_seed,
+                debug=False,
+                **kwargs):
+
+        self.experiment_kwargs = dict(
```

Comment:
This will be replaced by [tune.with_parameters](https://docs.ray.io/en/master/tune/api_docs/execution.html#tune-with-parameters) functionality in ray v1.1.0

Reply:
Actually should be available in 1.0.1 :) which should be available now!

Reply:
Perfect! Code updated.